Add profiles docs (#3231)
dlt-hub/dlt: docs/website/docs/hub/profiles.md

---
title: Profiles

> description: Manage environment-specific configurations and secrets in dlt and dltHub Workspace

Reviewer: ```suggestion
description: Manage environment-specific configurations and secrets in dltHub Workspace
```

keywords: [dltHub, profiles, workspace, configuration, secrets, environments]
---

# Profiles

Profiles in `dlt` define **environment-specific configurations and secrets**.

> They let you manage separate settings for development, testing, and production — all within the same codebase.

Reviewer: ```suggestion
They let you manage separate settings for development, testing, and production using the same codebase.
```
(heavy ChatGPT-coded hehe)

Each profile provides isolated configuration, credentials, and working directories, ensuring your pipelines are secure and environment-aware.

## Overview

A **profile** is a named configuration context that controls how and where your pipelines run.
Profiles are defined and managed through [**TOML files**](../general-usage/credentials) located in the `.dlt` directory.

Profiles let you:

* Securely manage credentials for multiple environments.
* Isolate pipeline state, configuration, and local data storage.
* Switch between environments without changing code.

## Enable the Workspace and Profiles

Profiles are part of the [dltHub Workspace](workspace) feature.
To use them, first install `dlt` with Workspace support:

```sh
pip install "dlt[workspace]"
```

> Initialize a project:

Author: it's not merged yet, so tests will fail, we can add it later
